### PR TITLE
feat: re-calculate epoch reward after halving

### PR DESCRIPTION
**Pull request by zmcNotafraid**
https://github.com/Magickbase/ckb-explorer-public-issues/issues/487

### CODE REVIEW

File: app/utils/ckb_utils.rb [hunk 1]
```diff
@@ -103,6 +103,16 @@ def self.base_reward(block_number, epoch_number)
     end
   end
 
+  def self.epoch_reward_with_halving(epoch_number)
+    if CkbSync::Api.instance.mode == "mainnet"
```
Comment (rabbitz) [inline app/utils/ckb_utils.rb:107]: "mainnet" 的常量为 CKB::MODE::MAINNET

Comment (quake) [reply app/utils/ckb_utils.rb:107]: halving is applied on testnet also.

File: app/utils/ckb_utils.rb [hunk 2]
```diff
@@ -103,6 +103,16 @@ def self.base_reward(block_number, epoch_number)
...
+      first_halving_epoch = 8760
+      radio = epoch_number / first_halving_epoch
+      Settings.default_epoch_reward.to_i / (2**radio)
+    else
+      Settings.default_epoch_reward.to_i
+    end
+  end
```
Comment (quake) [inline app/utils/ckb_utils.rb:114]:
```suggestion
  def self.epoch_reward_with_halving(epoch_number)
      Settings.default_epoch_reward >> epoch_number / 8760
  end
```

File: app/utils/ckb_utils.rb (hunk 1 above)
Comment (zmcNotafraid) [reply app/utils/ckb_utils.rb]: > halving is applied on testnet also.

So the testnet's halving epoch is also 8760 ?

Comment (quake) [reply app/utils/ckb_utils.rb]: yes

File: app/utils/ckb_utils.rb [hunk 3]
```diff
@@ -103,6 +103,11 @@ def self.base_reward(block_number, epoch_number)
     end
   end
 
+  def self.epoch_reward_with_halving(epoch_number)
+    first_halving_epoch = 8760
```
Comment (quake) [inline app/utils/ckb_utils.rb:107]: suggest to define a constant and add some code comment, e.g.

```
    // The block reward halves approximately every 4 years, one epoch is about 4 hours
    HALVING_EPOCH = 4 * 365 * 24 / 4
```

**Review by quake — APPROVED** [review]

**Pull request merged**